control-plane/go.mod

```diff
@@ -2,7 +2,10 @@ module github.com/hashicorp/consul-k8s/control-plane
 
 // TODO: remove when the SDK is released for Consul 1.17
 // The replace directive is needed be because `api` requires 0.14.1 of SDK and is both a direct and indirect dependency
-replace github.com/hashicorp/consul/sdk v0.14.1 => github.com/hashicorp/consul/sdk v0.4.1-0.20230825164720-ecdcde430924
+replace (
+	github.com/hashicorp/consul/proto-public v0.4.1 => github.com/hashicorp/consul/proto-public v0.1.2-0.20230911164019-a69e901660bd
```

Comment:
is there a reason why we're adding a new replace here? if it's to work with an unreleased version should we update this comment as well?

Reply:
I added a comment explaining the replace directive that I added as well as making the TODO more robust